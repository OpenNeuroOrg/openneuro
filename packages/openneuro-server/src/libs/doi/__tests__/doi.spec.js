import { formatBasicAuth } from '../index.js'

describe('DOI minting utils', () => {
  describe('auth()', () => {
    it('returns a base64 basic auth string', () => {
      const doiConfig = { username: 'test', password: '12345' }
      expect(formatBasicAuth(doiConfig)).toBe('Basic dGVzdDoxMjM0NQ==')
    })
  })
})
