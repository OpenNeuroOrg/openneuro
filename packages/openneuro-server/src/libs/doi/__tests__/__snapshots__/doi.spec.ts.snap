// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DOI minting utils > template() > accepts expected arguments 1`] = `
"<?xml version="1.0" encoding="UTF-8"?>
<resource xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://datacite.org/schema/kernel-4" xsi:schemaLocation="http://datacite.org/schema/kernel-4 http://schema.datacite.org/meta/kernel-4/metadata.xsd">
  <identifier identifierType="DOI">12345</identifier>
  <creators>
  <creator><creatorName>A. User</creatorName></creator><creator><creatorName>B. User</creatorName></creator>
  </creators>
  <titles>
    <title xml:lang="en-us">Test Dataset</title>
  </titles>
  <publisher>Openneuro</publisher>
  <publicationYear>1999</publicationYear>
  <resourceType resourceTypeGeneral="Dataset">fMRI</resourceType>
</resource>"
`;
